### ISSUE TITLE
thread panic at lexer/mod.rs when running `no-constant-binary-expression`

### ISSUE THREAD

**Issue by Nsttt** [oxc-project/oxc#2095]
Hello ! Not sure if this is relevant, but might help fix some possible bugs.

I was following [this tweet](https://twitter.com/boshen_c/status/1736029429896548821) from a month ago on some repos and making some PRs. But apparently the [effect](https://github.com/Effect-TS/effect) repo has something that makes oxlint die when I ran `npx oxlint@latest -D no-constant-binary-expression`.

I'm not versed on Rust _yet_, so can't really tell what's happening but I've attached a backtrace below that may provide some insight.

```
❯ RUST_BACKTRACE=full npx oxlint@latest -D no-constant-binary-expression
thread '<unnamed>' panicked at crates/oxc_parser/src/lexer/mod.rs:339:40:
no entry found for key
stack backtrace:
   0:        0x1052b9728 - __mh_execute_header
   1:        0x10500a1bc - __mh_execute_header
   2:        0x10529a66c - __mh_execute_header
   3:        0x1052bd30c - __mh_execute_header
   4:        0x1052bcc04 - __mh_execute_header
   5:        0x1052bdad8 - __mh_execute_header
   6:        0x1052bd64c - __mh_execute_header
   7:        0x1052bd5b8 - __mh_execute_header
   8:        0x1052bd5ac - __mh_execute_header
   9:        0x105323a54 - __mh_execute_header
  10:        0x105323e14 - __mh_execute_header
  11:        0x105192690 - __mh_execute_header
  12:        0x105198f34 - __mh_execute_header
  13:        0x105198d20 - __mh_execute_header
  14:        0x105198588 - __mh_execute_header
  15:        0x105198674 - __mh_execute_header
  16:        0x10519878c - __mh_execute_header
  17:        0x1051a81f8 - __mh_execute_header
  18:        0x105195228 - __mh_execute_header
  19:        0x105193f14 - __mh_execute_header
  20:        0x105127454 - __mh_execute_header
  21:        0x105124d0c - __mh_execute_header
  22:        0x105125288 - __mh_execute_header
  23:        0x10533cf94 - __mh_execute_header
  24:        0x1051e6e38 - __mh_execute_header
  25:        0x1051e6b88 - __mh_execute_header
  26:        0x1052beb90 - __mh_execute_header
  27:        0x189b71034 - __pthread_joiner_wake
```

**Comment by Boshen** [comment]
This is fixed in https://github.com/oxc-project/oxc/pull/2089

I'll make a release soon.

**Comment by Boshen** [comment]
Should be fixed in v0.2.2. I've tested it in the `effect` repo.
